File: LeetCode-Cpp/BasicCalculator.cpp
class Solution {
public:
    int calculate(string s) {
        stack<int> nums;  // sayıları tutacak yığın
        stack<char> ops;  // operatörleri tutacak yığın
        int num = 0;
        int result = 0;
        int sign = 1;  // '+' işareti için başlangıçta pozitif

        for (int i = 0; i < s.length(); i++) {
            char c = s[i];

            if (isdigit(c)) {
                num = num * 10 + (c - '0');  // sayıyı birleştir
            } else if (c == '+' || c == '-') {
                result += sign * num;  // bir önceki sayıyı işaretine göre ekle
                num = 0;               // yeni sayıya başla
                sign = (c == '+') ? 1 : -1;  // işareti güncelle
            } else if (c == '(') {
                // parantez açıldığında önceki sonucu ve işareti kaydet
                nums.push(result);
                ops.push(sign);
                result = 0;
                sign = 1;
            } else if (c == ')') {
                result += sign * num;  // iç parantezdeki sonucu hesapla
                num = 0;
                result *= ops.top();  // işareti uygula
                ops.pop();
                result += nums.top();  // önceki sonucu ekle
                nums.pop();
            }
        }
        result += sign * num;  // son kalan sayıyı ekle
        return result;
        }
};

File: LeetCode-Python/BasicCalculator.py
class Solution:
    def calculate(self, s: str) -> int:
        stack = []
        current_number = 0
        result = 0
        sign = 1  # 1: pozitif, -1: negatif

        for char in s:
            if char.isdigit():
                current_number = current_number * 10 + int(char)  # Sayıyı oluştur
            elif char in "+-":
                result += sign * current_number  # Geçerli sayıyı sonuçla ekle
                current_number = 0  # Sayıyı sıfırla
                sign = 1 if char == '+' else -1  # İşlemi güncelle
            elif char == '(':
                stack.append(result)  # Geçerli sonucu yığının üstüne ekle
                stack.append(sign)  # İşlemi yığının üstüne ekle
                result = 0  # Sonucu sıfırla
                sign = 1  # Yeni alt ifade için işaret sıfırla
            elif char == ')':
                result += sign * current_number  # Geçerli sayıyı ekle
                current_number = 0  # Sayıyı sıfırla
                result *= stack.pop()  # İşaretin etkisini uygula
                result += stack.pop()  # Yığından önceki sonucu ekle

        # En son sayıyı ekle
        result += sign * current_number
        return result
